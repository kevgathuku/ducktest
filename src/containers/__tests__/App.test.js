import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { App } from '../App'
import { Picker } from '../../components/Picker'

describe('App', () => {
  let props

  beforeEach(() => {
    props = {
      isFetching: undefined,
      dispatch: undefined,
      posts: undefined,
      selectedSubreddit: undefined
    }
  })

  it('renders without crashing given the required props', () => {
    props = {
      isFetching: false,
      dispatch: jest.fn(),
      selectedSubreddit: 'reactjs',
      posts: []
    }

    const wrapper = shallow(<App {...props} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('always renders the Picker component', () => {
    props = {
      isFetching: false,
      dispatch: jest.fn(),
      selectedSubreddit: 'reactjs',
      posts: []
    }

    const wrapper = shallow(<App {...props} />)
    expect(wrapper.find(Picker)).toBeTruthy()
  })

  it('sets the selectedSubreddit prop as the `value` prop on the Picker component', () => {
    props = {
      isFetching: false,
      dispatch: jest.fn(),
      selectedSubreddit: 'reactjs',
      posts: []
    }

    const wrapper = shallow(<App {...props} />)
    const PickerComponent = wrapper.find(Picker)

    expect(PickerComponent.props().value).toBe(props.selectedSubreddit)
  })

  it('renders the Refresh button when the isFetching prop is false', () => {
    props = {
      isFetching: false,
      dispatch: jest.fn(),
      selectedSubreddit: 'reactjs',
      posts: []
    }

    const wrapper = shallow(<App {...props} />)

    expect(wrapper.find('button').length).toBe(1)
  })
})
