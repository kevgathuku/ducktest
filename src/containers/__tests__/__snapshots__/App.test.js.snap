// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<div>
  <Picker
    onChange={[Function]}
    options={
      Array [
        "reactjs",
        "frontend",
      ]
    }
    value="reactjs"
  />
  <p>
    <button
      onClick={[Function]}
    >
      Refresh
    </button>
  </p>
  <h2>
    Empty.
  </h2>
</div>
`;
